src/lib/frontend/models.ml

```diff
@@ -238,7 +242,7 @@ module SmtlibCounterExample = struct
 
   let pp_abstract_value_of_type ppf ty =
     if not @@ Options.get_interpretation_use_underscore () then
-      Fmt.pf ppf "(as @@%s %a)" (Hstring.fresh_string ()) Ty.pp_smtlib ty
+      Fmt.pf ppf "(as @@a%i %a)" !fresh_counter Ty.pp_smtlib ty
```

Comment:
This needs to increment the counter, otherwise this will always be `a0`.

Reply:
oupsi! We should have a test with two abstract values :p